File: Sort/QuickSort.java
/**
 * 快速排序算法
 */

public class QuickSort {
    public static void sort(int[] arr, int l, int r) {
        if (l < r) {
            int i = l, j = r;
            int axis = arr[l];
            while (i < j) {
                while (i < j && arr[j] >= axis) j--;
                if (i < j) arr[i++] = arr[j];

                while (i < j && arr[i] < axis) i++;
                if (i < j) arr[j--] = arr[i];
            }

            arr[i] = axis;
            sort(arr, l, i - 1);
            sort(arr, i + 1, r);
        }
    }

    public static void main(String[] args) {
        int[] arr = {3,1,5,2,4,9,0};
        sort(arr, 0, arr.length - 1);
        for (int i = 0; i < arr.length; i++) {
            System.out.println(arr[i]);
        }
    }

}

File: Sort/quick_sort.py

def quick_sort(nums):

    def sort(nums, l, r):

        if l >= r:
            return

        x = nums[l]
        i, j = l, r
        while i < j:

            while i < j and nums[j] > x:
                j -= 1
            if i < j:
                nums[i] = nums[j]
                i += 1
            while i < j and nums[i] <= x:
                i += 1
            if i < j:
                nums[j] = nums[i]
                j -= 1

        nums[i] = x
        sort(nums, l, i - 1)
        sort(nums, i + 1, r)

    sort(nums, 0, len(nums) - 1)

    return nums

if __name__ == "__main__":

    print(quick_sort([72, 6, 57, 88, 60, 42, 83, 73, 48, 85]))